response_operations_ui/views/respondents.py

```diff
@@ -50,4 +50,91 @@ def respondent_details(respondent_id):
+@login_required
+def view_contact_details(respondent_id):
+    respondents_details = party_controller.get_respondent_by_party_id(respondent_id)
+
+    form = EditContactDetailsForm(form=request.form, default_values=respondents_details)
+
+    return render_template('edit-contact-details.html', respondent_details=respondents_details,
+                           form=form, tab='respondents')
+
+
+@respondent_bp.route('/edit-contact-details/<respondent_id>', methods=['POST'])
+@login_required
+def edit_contact_details(respondent_id):
+    form = request.form
+    contact_details_changed = party_controller.update_contact_details(respondent_id, form)
+
+    ui_message = 'No updates were necessary'
+    if 'emailAddress' in contact_details_changed:
+        ui_message = f'Contact details changed and verification email sent to {form.get("email")}'
+    elif len(contact_details_changed) > 0:
+        ui_message = 'Contact details changed'
+
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=respondent_id, info=ui_message))
+
+
+@respondent_bp.route('/resend_verification/<respondent_id>', methods=['GET'])
+@login_required
+def view_resend_verification(respondent_id):
+    logger.debug("Re-send verification email requested", respondent_id=respondent_id)
+    respondent = party_controller.get_respondent_by_party_id(respondent_id)
+    email = respondent['pendingEmailAddress'] if 'pendingEmailAddress' in respondent \
+        else respondent['emailAddress']
+
+    return render_template('re-send-verification-email.html', respondent_id=respondent_id, email=email,
+                           tab='respondents')
+
+
+@respondent_bp.route('/resend_verification/<party_id>', methods=['POST'])
+@login_required
+def resend_verification(party_id):
+    reporting_units_controllers.resend_verification_email(party_id)
+    logger.info("Re-sent verification email.", party_id=party_id)
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=party_id,
+                            info='Verification email re-sent'))
+
+
+@respondent_bp.route('<respondent_id>/change-enrolment-status', methods=['POST'])
+@login_required
+def change_enrolment_status(respondent_id):
+
+    reporting_units_controllers.change_enrolment_status(business_id=request.args['business_id'],
+                                                        respondent_id=respondent_id,
+                                                        survey_id=request.args['survey_id'],
+                                                        change_flag=request.args['change_flag'])
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=respondent_id,  enrolment_changed='True'))
+
+
+@respondent_bp.route('/<ru_ref>/change-enrolment-status', methods=['GET'])
+@login_required
+def confirm_change_enrolment_status(ru_ref):
+    return render_template('confirm-enrolment-change.html', business_id=request.args['business_id'], ru_ref=ru_ref,
```

Comment:
This feels wrong to me, but I'm not sure what 'right' looks like.  I assume the url for this looks like `<host>/<ru_ref>/change-enrolment-status?ru_name=blah&trading_as=blah.......`  If that's right, then it seems like such a crazy long url and unnecessary.  Though maybe I'm wrong in thinking that,  I'd ask others to see what they think.

Reply:
We can ignore this for now.  I don't know enough to offer an alternative, and there are other things that need doing instead